### PR TITLE
feat: operator state after update curve tests

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: test/CSModule.t.sol [hunk 1]
```diff
@@ -7224,6 +7224,297 @@ contract CSMDepositableValidatorsCount is CSMCommon {
...
+
+        assertEq(depositableBefore, 0);
+
+        updateToBetterCurve();
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        csm.normalizeQueue(noId);
```
Comment (dgusakov) [inline test/CSModule.t.sol:7255]: Not needed here

File: test/CSModule.t.sol [hunk 2]
```diff
@@ -7224,6 +7224,297 @@ contract CSMDepositableValidatorsCount is CSMCommon {
...
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
```
Comment (dgusakov) [inline test/CSModule.t.sol:7281]: It's also not needed. Unbonded assertion is missing

File: test/CSModule.t.sol (hunk 1 above)
Comment (dgusakov) [reply test/CSModule.t.sol:7255]: Replace with unbonded = 0 assertion

File: test/CSModule.t.sol [hunk 3]
```diff
@@ -7224,6 +7224,297 @@ contract CSMDepositableValidatorsCount is CSMCommon {
...
```
Comment (dgusakov) [inline test/CSModule.t.sol:7355]: I could not get the idea here. I think a test when unbonded becomes bonded makes more sense. WDYT?

File: test/CSModule.t.sol [hunk 4]
```diff
@@ -7224,6 +7224,297 @@ contract CSMDepositableValidatorsCount is CSMCommon {
...
```
Comment (dgusakov) [inline test/CSModule.t.sol:7386]: Add unbonded assertions. Not sure if normalizing queue when all are bonded makes sense

**Review by dgusakov — CHANGES_REQUESTED** [review]
Pls review assertions in general. I guess some are useless, while some can be added

**Review by dgusakov — DISMISSED** [review]

File: test/CSModule.t.sol [hunk 5]
```diff
@@ -7224,6 +7224,424 @@ contract CSMDepositableValidatorsCount is CSMCommon {
     }
 }
 
+contract CSMNodeOperatorStateAfterUpdateCurve is CSMCommon {
```
Comment (skhomuti) [inline test/CSModule.t.sol:7227]: add invariant modifiers pls

File: test/CSModule.t.sol [hunk 6]
```diff
@@ -7224,6 +7224,424 @@ contract CSMDepositableValidatorsCount is CSMCommon {
...
+            requiredBefore,
+            "Required bond should increase"
+        );
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore,
+            "Depositables should not change after curve update"
+        );
+        assertEq(
+            accounting.getUnbondedKeysCountToEject(noId),
```
Comment (skhomuti) [inline test/CSModule.t.sol:7359]: there is no lock, so i'd expect more straightforward `getUnbondedKeysCount` here

**Review by skhomuti — CHANGES_REQUESTED** [review]

File: test/CSModule.t.sol (hunk 5 above)
Comment (vgorkavenko) [reply test/CSModule.t.sol:7227]: done

File: test/CSModule.t.sol (hunk 6 above)
Comment (vgorkavenko) [reply test/CSModule.t.sol]: done

**Review by dgusakov — APPROVED** [review]

**Pull request merged**